### PR TITLE
feat(WebWorker): Expose MessageBroker API

### PR DESCRIPTION
**Pull request by jteplitz**

### CODE REVIEW

File: modules/angular2/src/web_workers/ui/application.dart [hunk 1]
```diff
@@ -14,24 +14,25 @@ import 'package:angular2/src/web_workers/shared/isolate_message_bus.dart';
...
+    instance.app = bootstrapUICommon(instance.bus);
+    return instance;
   });
 }
 
 /**
  * To be called from the main thread to spawn and communicate with the worker thread
  */
-Future<MessageBus> spawnWebWorker(Uri uri) {
+Future<IsolateInstance> spawnWebWorker(Uri uri) {
```
Comment (yjbanov) [inline modules/angular2/src/web_workers/ui/application.dart]: You could use `async/await` to simplify the code here:

```dart
IsolateInstance spawnWebWorker(Uri uri) async {
  var receivePort = new ReceivePort();
  var isolateEndSendPort = receivePort.sendPort;
  var isolate = await Isolate.spawnUri(uri, const [], isolateEndSendPort);
  var source = new UIMessageBusSource(receivePort);
  var sendPort = await source.sink;
  var sink = new IsolateMessageBusSink(sendPort);
  var bus = new IsolateMessageBus(sink, source);
  return new IsolateInstance(null, isolate, bus);
}
```

File: modules/angular2/src/web_workers/ui/application.dart [hunk 2]
```diff
@@ -43,3 +44,11 @@ class UIMessageBusSource extends IsolateMessageBusSource {
         return message is SendPort;
       });
 }
+
+class IsolateInstance {
```
Comment (yjbanov) [inline modules/angular2/src/web_workers/ui/application.dart]: The name of this class is not self-descriptive. Let's add some doc comments about why it is needed.

File: modules/angular2/src/web_workers/ui/application.dart [hunk 3]
```diff
@@ -43,3 +44,11 @@ class UIMessageBusSource extends IsolateMessageBusSource {
...
+  WebWorkerApplication app;
+  Isolate isolate;
+  MessageBus bus;
```
Comment (yjbanov) [inline modules/angular2/src/web_workers/ui/application.dart]: Could the fields be `final`?

File: modules/angular2/src/web_workers/ui/impl.ts [hunk 4]
```diff
@@ -15,25 +15,43 @@ import {WebWorkerSetup} from 'angular2/src/web_workers/ui/setup';
...
+          var zone = createNgZone();
+          wtfInit();
+          return zone.run(() => {
+            var injector = createInjector(zone, bus);
+            // necessary to kick off all the message based components
+            return injector.get(WebWorkerApplication);
+          });
+        }
+
+@Injectable() export class WebWorkerApplication {
```
Comment (yjbanov) [inline modules/angular2/src/web_workers/ui/impl.ts]: We normally place `@Injectable` annotation on the previous line.

File: modules/examples/e2e_test/web_workers/message_broker/message_broker_spec.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  afterEach(verifyNoBrowserErrors);
+
+  it('should bootstrap', () => {
+    browser.get(URL);
+    waitForBootstrap();
+    expect(element(by.css("app h1")).getText()).toEqual("WebWorker MessageBroker Test");
+  });
+
+  it("should echo messages", () => {
```
Comment (yjbanov) [inline modules/examples/e2e_test/web_workers/message_broker/message_broker_spec.ts]: for consistency let's use either `'` or `"` but not both

Comment (jteplitz602) [inline modules/examples/e2e_test/web_workers/message_broker/message_broker_spec.ts]: Not sure why github hasn't marked this as an outdated diff, but I changed all the ```'```s to ```"```s

**Pull request closed without merging**